### PR TITLE
chore: Formatted with stylua.toml and added stylua.toml

### PR DESCRIPTION
**Pull request by ashincoder**

### CODE REVIEW

**Comment by G-Rowell** [comment]
This is a much cleaner PR now!
Let me take a look before I approve :)

**Comment by ashincoder** [comment]
yes do please :D

File: lua/plugins/bufferline.lua
Comment (G-Rowell) [inline lua/plugins/bufferline.lua:86]: Hmmmmm, it looks like the stylua config has a super duper big indentation.

Maybe it should be smaller?

File: stylua.toml
Comment (G-Rowell) [inline stylua.toml:1]: These look like good settings I think (though I'm no expert at stylua).

I like this

**Review by G-Rowell — COMMENTED** [review]
Hmmmm, 
Looks good so far, 
except that it looks like the indentation is super super big? (or this just from viewing on GitHub?)

Also, can you guarantee that the only manual change you've done is adding the `stylua.toml` file?

Also, I think stylua can be a hidden file? `.stylua.toml` instead?
Can you check the stylua docs for this? I've seen plenty of projects have it as a hidden file

File: stylua.toml
Comment (ashincoder) [reply stylua.toml:1]: Great

File: lua/plugins/bufferline.lua
Comment (ashincoder) [reply lua/plugins/bufferline.lua:86]: I think the big indentation is from github. It looks good to me. I shall rename it to .stylua.toml then

File: .stylua.toml
Comment (G-Rowell) [inline .stylua.toml:1]: I'm being a little pendandtic here, but I'd love for you to learn this skill.

Let's remove this commit to just change the filename and instead edit the original commit
`git rebase --interactive HEAD~3`
- Find the latest commit that just changes the name,
- change from `pick` to `drop` to remove this 3rd commit.
- change the commit which creates the `stylua.toml` file, from `pick` to `edit`

Follow the prompts, but this will drop you into editing this commit, 
you can rename the file `mv stylua.toml .stylua.toml`, then git add like normal, follow the git prompts.

At the end you should have a cleaner commit log check with `git log`

BUT you will not be able to push to your branch because you have edited your tree, 
so you will need to `git push origin --force` force a push.

Then comment back in this PR to let me know

**Review by G-Rowell — CHANGES_REQUESTED** [review]

File: .stylua.toml
Comment (ashincoder) [reply .stylua.toml:1]: I think I am going to die xD. I think this enough for now. I did what u said. But... are u using github-cli

Comment (G-Rowell) [reply .stylua.toml:1]: No, just the normal `git` on the command line

What is your workflow for pulling & pushing changes?

**Comment by siduck76** [comment]
@G-Rowell is this ready?

**Comment by siduck76** [comment]
@ashincoder please format the code locally too first , I cloned your fork and I see this 

image

**Comment by G-Rowell** [comment]
> @G-Rowell is this ready?

Click 'Show resolved' on the review right above your commit.

It is ready but the there is a redundant commit just to rename the file (instead of just editing the original commit).

You can merge this but I think we should aim to reduce commits like #27cf00e 
as this commit does nothing but rename a file.

So you can merge if you like, but @ashincoder would really benefit from learning how to edit a commit, it's a pretty useful & v. common thing to do.
If you do merge before cleaning this up it'll be yet another thing I'll need to clean from our git repo at some point

**Comment by siduck76** [comment]
after formatting it myself (with stylua)

image

**Comment by G-Rowell** [comment]
NOTE: not ready per my updated comment above

**Comment by siduck76** [comment]
@G-Rowell I dont see anything formatted , where is it :/

**Comment by G-Rowell** [comment]
Hold on a few minutes, diagnosing this rn
Don't merge this yet

**Comment by ashincoder** [comment]
This is ready. Sorry for the late reply. There is a an I internet problem
here.

**Comment by siduck76** [comment]
> Hold on a few minutes, diagnosing this rn
> Don't merge this yet

test it locally and you'll see the change ( i see nothing formatted , it breaks the indentline very bad)

**Comment by ashincoder** [comment]
Guys I am using my own branch it works fine

**Comment by ashincoder** [comment]
I have formatted locally it just like that

**Comment by siduck76** [comment]
This PR's formatted code : 

image

I used the same stylua.toml and formatted it locally , which looks like this 

image

@ashincoder @G-Rowell

**Comment by siduck76** [comment]
did you formatting the code locally or is it some github ci or such thing formatting this thing :/ ?

**Comment by ashincoder** [comment]
I think it is because I used the command stylua in my machine to format. But neoformat uses it's stylua config I think

**Comment by ashincoder** [comment]
> did you formatting the code locally or is it some github ci or such thing formatting this thing :/ ?

I formatted locally with stylua that is installed in my machine

**Comment by siduck76** [comment]
> I think it is because I used the command stylua in my machine to format. But neoformat uses it's stylua config I think

use the stylelua.toml in this PR , rename it as ~/.stylelua.toml ( this worked for me)

**Comment by ashincoder** [comment]
[asciicast](https://asciinema.org/a/NJ76twItpMDuM0Q9IaqbtpyI2)

**Comment by ashincoder** [comment]
> > I think it is because I used the command stylua in my machine to format. But neoformat uses it's stylua config I think
> 
> use the stylelua.toml in this PR , rename it as ~/.stylelua.toml ( this worked for me)

I see the change

**Comment by siduck76** [comment]
@aserowy @G-Rowell has sent an explanation about why not merging in this PR in the discord server , read it. check git rebase and make the PR commits cleaner :heart:

File: .stylua.toml
Comment (ashincoder) [reply .stylua.toml]: I just do `git pull ` and `git push ` in my command line

**Pull request closed without merging**
